server/src/main/java/io/crate/planner/optimizer/symbol/Optimizer.java

```diff
@@ -132,14 +135,22 @@ public Symbol tryApplyRules(Symbol node) {
     }
 
     private class Visitor extends FunctionCopyVisitor<Void> {
+        private Stack<io.crate.expression.symbol.Function> visitedFunctions = new Stack<>();
 
         @Override
         public Symbol visitFunction(io.crate.expression.symbol.Function symbol, Void context) {
+            visitedFunctions.push(symbol);
             var maybeTransformedSymbol = tryApplyRules(symbol);
             if (symbol.equals(maybeTransformedSymbol) == false) {
                 return maybeTransformedSymbol;
             }
```

Comment:
If this exists here, `visitedFunctions.pop` is not called and the symbol will stay in the stack.